adapters/grid/grid.go

```diff
@@ -69,16 +97,224 @@ func setImpExtData(imp openrtb2.Imp) openrtb2.Imp {
 	return imp
 }
 
+func mixKeywords(keywordsString string, keywords map[string]interface{}) {
+	var ortb2Array []interface{}
+
+	keywordsArr := strings.Split(keywordsString, ",")
+
+	if len(keywordsArr) > 0 {
+		keywordsInt := make([]interface{}, len(keywordsArr))
+		for i, v := range keywordsArr {
+			keywordsInt[i] = v
+		}
+		ortb2Keywords := map[string]interface{}{
+			"name":     "keywords",
+			"keywords": keywordsInt,
+		}
+		if keywords["ortb2"] == nil {
+			ortb2Array = make([]interface{}, 0)
+		} else {
+			ortb2Array = keywords["ortb2"].([]interface{})
+		}
+		ortb2Array = append(ortb2Array, ortb2Keywords)
+		keywords["ortb2"] = ortb2Array
+	}
+}
+
+func mergeWithReqExtKeywords(extKeywords map[string]interface{}, request *openrtb2.BidRequest) {
+	var reqExt ReqExt
+	var reqExtKeywords map[string]interface{}
+
+	if err := json.Unmarshal(request.Ext, &reqExt); err == nil {
+		if reqExt.Keywords != nil {
+			json.Unmarshal(reqExt.Keywords, &reqExtKeywords)
+
+			for key, keyword := range reqExtKeywords {
+				if extKeywords[key] == nil {
+					extKeywords[key] = keyword
+				} else {
+					if key == "site" || key == "user" {
+						target := extKeywords[key].(map[string]interface{})
+						from := keyword.(map[string]interface{})
+
+						for name, value := range from {
+							if target[name] == nil {
+								target[name] = value
+							} else {
+								valueArr := value.([]interface{})
+								targetArr := target[name].([]interface{})
+								target[name] = append(targetArr, valueArr...)
+							}
+						}
+					} else {
+						extKeywords[key] = keyword
+					}
+				}
+			}
+		}
+	}
+}
+
+func reformatExtKeywords(extKeywords map[string]interface{}) {
```

Comment:
Can we refactor the `reformatExtKeywords(extKeywords map[string]interface{})` function or subdivide it into subfunctions to make it easier to read?